src/Message/PaymentIntents/AuthorizeRequest.php

```diff
@@ -0,0 +1,374 @@
+     */
+    public function setApplicationFee($value)
+    {
+        return $this->setParameter('applicationFee', $value);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getStatementDescriptor()
+    {
+        return $this->getParameter('statementDescriptor');
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return AbstractRequest provides a fluent interface.
+     */
+    public function setStatementDescriptor($value)
+    {
+        $value = str_replace(array('<', '>', '"', '\''), '', $value);
+
+        return $this->setParameter('statementDescriptor', $value);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getReceiptEmail()
+    {
+        return $this->getParameter('receipt_email');
+    }
+
+    /**
+     * @param mixed $email
+     * @return $this
+     */
+    public function setReceiptEmail($email)
+    {
+        $this->setParameter('receipt_email', $email);
+
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getData()
+    {
+        $this->validate('amount', 'currency');
+
+        $data = array();
+
+        $data['amount'] = $this->getAmountInteger();
+        $data['currency'] = strtolower($this->getCurrency());
+        $data['description'] = $this->getDescription();
+        $data['metadata'] = $this->getMetadata();
+
+        if ($this->getStatementDescriptor()) {
+            $data['statement_descriptor'] = $this->getStatementDescriptor();
+        }
+        if ($this->getDestination()) {
+            $data['transfer_data']['destination'] = $this->getDestination();
+        }
+
+        if ($this->getOnBehalfOf()) {
+            $data['on_behalf_of'] = $this->getOnBehalfOf();
+        }
+
+        if ($this->getApplicationFee()) {
+            $data['application_fee'] = $this->getApplicationFeeInteger();
+        }
+
+        if ($this->getTransferGroup()) {
+            $data['transfer_group'] = $this->getTransferGroup();
+        }
+
+        if ($this->getReceiptEmail()) {
+            $data['receipt_email'] = $this->getReceiptEmail();
+        }
+
+        if ($this->getPaymentMethod()) {
+            $data['payment_method'] = $this->getPaymentMethod();
+        } elseif ($this->getSource()) {
+            $data['payment_method'] = $this->getSource();
+        } elseif ($this->getCardReference()) {
+            $data['payment_method'] = $this->getCardReference();
+        } elseif ($this->getToken()) {
+            $data['payment_method_data'] = [
+                'type' => 'card',
+                'card' => ['token' => $this->getToken()],
+            ];
+        } else {
+            // one of cardReference, token, or card is required
+            $this->validate('paymentMethod');
+        }
+
+        if ($this->getCustomerReference()) {
+            $data['customer'] = $this->getCustomerReference();
+        }
+
+        $data['confirmation_method'] = 'manual';
+        $data['capture_method'] = 'manual';
```

Comment:
@andris-sevcenko Any reason why I can't set the capture_method to **automatic** like documented here: 
https://stripe.com/docs/api/payment_intents/create#create_payment_intent-capture_method

Right now I'm missing a way to capture my done payment for PaymentIntent gateway

Reply:
You can call the `completePayment()` method to complete the payment as that's how it's done for offsite Omnipay gateways.

Setting `automatic` didn't make much sense at the time, because automatic payment intents can be confirmed only using Stripe.js.

This PR by no means was perfect, but probably the best way to address its shortcomings is to open a PR that improves them :)

Reply:
> Setting automatic didn't make much sense at the time, because automatic payment intents can be confirmed only using Stripe.js.
Exactly - AFAIK for the "automatic" you need to handle asynchronous Stripe Webhook

https://stripe.com/docs/payments/payment-intents/migration/automatic-confirmation#saving-cards-checkout-step-4

> Automatic confirmation is recommended if you are already using webhooks with your existing payments integration.
> If you are migrating from an existing cards and Charges API integration, we recommend using manual confirmation, which is the fastest way to migrate.

vs:
https://stripe.com/docs/payments/payment-intents/migration

Reply:
ok, convinced! :)
But the readme file should explain that capture has to be called in my point of view.
After confirmation I do call now \Omnipay\Stripe\Message\PaymentIntents\CaptureRequest.

completePayment() is also not mentioned, only confirm(), but honestly I didn't check the difference. :)